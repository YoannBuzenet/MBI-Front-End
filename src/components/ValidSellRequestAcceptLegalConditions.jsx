import React from "react";

const ValidSellRequestAcceptLegalConditions = props => {
  return (
    <div className="isUserAuthenticated">
      <form action="" onSubmit={props.handleSubmit}>
        <input type="checkbox" required id="checkbox-compliance" />
        <label htmlFor="checkbox-compliance">
          J'accepte les conditions générales de ventes.
        </label>
        <button className="sellRequest-validation-button" type="submit">
          Valider mon rachat
        </button>
      </form>
    </div>
  );
};

export default ValidSellRequestAcceptLegalConditions;
