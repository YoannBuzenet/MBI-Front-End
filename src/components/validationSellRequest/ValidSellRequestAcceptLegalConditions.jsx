import React from "react";
import { FormattedMessage } from "react-intl";

const ValidSellRequestAcceptLegalConditions = (props) => {
  return (
    <div className="isUserAuthenticated">
      <form action="" onSubmit={props.handleSubmit}>
        <input type="checkbox" required id="checkbox-compliance" />
        <label htmlFor="checkbox-compliance">
          <FormattedMessage
            id="app.sellRequestValidation.legalConditionsAccepting"
            defaultMessage={`I accept the shop buying clauses.`}
          />
        </label>
        <button className="sellRequest-validation-button" type="submit">
          <FormattedMessage
            id="app.sellRequestValidation.acceptCTA"
            defaultMessage={`Validate`}
          />
        </button>
      </form>
    </div>
  );
};

export default ValidSellRequestAcceptLegalConditions;
