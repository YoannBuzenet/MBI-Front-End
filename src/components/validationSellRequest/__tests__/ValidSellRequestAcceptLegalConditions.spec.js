import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ValidSellRequestAcceptLegalConditions from "../ValidSellRequestAcceptLegalConditions";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ValidSellRequestAcceptLegalConditions />, div);
});

it("renders its UI correctly", () => {
  const validationComponent = renderer
    .create(<ValidSellRequestAcceptLegalConditions />)
    .toJSON();
  expect(validationComponent).toMatchSnapshot();
});
