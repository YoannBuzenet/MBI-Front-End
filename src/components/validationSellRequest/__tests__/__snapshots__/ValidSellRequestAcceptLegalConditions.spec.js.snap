// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General tests renders its UI correctly 1`] = `
<div
  className="isUserAuthenticated"
>
  <form
    action=""
  >
    <input
      id="checkbox-compliance"
      required={true}
      type="checkbox"
    />
    <label
      htmlFor="checkbox-compliance"
    >
      I accept the 
      <a
        href="/buyingClauses"
        target="_blank"
      >
        shop buying clauses
      </a>
      .
    </label>
    <button
      className="sellRequest-validation-button"
      type="submit"
    >
      Validate
    </button>
  </form>
</div>
`;
